kong/plugins/rate-limiting/policies/init.lua

```diff
@@ -112,6 +134,49 @@ local function get_redis_connection(conf)
 end
 
 
+local function sync_to_redis(premature, conf)
+  if premature then
+    return
+  end
+
+  local red, err = get_redis_connection(conf)
+  if not red then
+    kong.log.err("[rate-limiting] failed to connect to Redis: ", err)
+    return
+  end
+
+  red:init_pipeline()
+
+  for cache_key, delta in pairs(cur_delta) do
+    red:eval([[
+      local key, value, expiration = KEYS[1], tonumber(ARGV[1]), ARGV[2]
+      local exists = redis.call("exists", key)
+      redis.call("incrby", key, value)
+      if not exists or exists == 0 then
+        redis.call("expireat", key, expiration)
+      end
+    ]], 1, cache_key, delta, cur_usage_expire_at[cache_key])
+  end
+
+  local _, err = red:commit_pipeline()
+  if err then
+    kong.log.err("[rate-limiting] failed to commit increment pipeline in Redis: ", err)
+    return
+  end
+
+  local ok, err = red:set_keepalive(10000, 100)
+  if not ok then
+    kong.log.err("[rate-limiting] failed to set Redis keepalive: ", err)
+    return
```

Comment:
we should clear tables before exiting this function, or use `goto failed`.

Reply:
done